### PR TITLE
chore: add visual tests setup for /viewer 

### PR DESCRIPTION
**Pull request by christjt**
# Description

This includes a setup to run and create visual tests within the context of /viewer.

Test by running `yarn run test:visual` in /viewer.

New capabilities:

- Run visual tests with a single command `yarn run test:visual`, which both serves a development server that hosts the tests, and runs screenshot tests using puppeteer.
- All tests are run within the same context / site, so overhead of running each test is considerably lower than previous solution.
- Watch mode is partially supported. I say partially because it requires the dev server and tests to be run in different processes. To do this run `yarn run test:visual:server` and `yarn run test:visual:client --watch` in two different terminal instances. In addition to this, the test client (the one that runs jest/puppeteer) doesn't actually listen to the visual test apps, so you need to trigger a run using the cli (press ENTER key). The gain is that the feedback loop is substantially lower, and re-running a test in watch mode usually only takes a few seconds.

To register a new visual test, it has to be registered in /viewer/packages/visualTests.root.ts (though I am experimenting with webpack's ability to do dynamic loading to try to overcome this).

There are also a naming convention which must be kept when creating a new visual test which is: `[MyTest].VisualTestFixture.ts` and the class name must be `[MyTest]VisualTestFixture`.

NOTE:
LGTM and code analyser complains about remote execution logic going on, but this is intended behaviour because we want to remotely execute code that is being served on the test server from the test runner. 

# Checklist:

Here is a checklist that should completed before merging this given feature. 
Any shortcomings from the items below should be explained and detailed within the contents of this PR.

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [x] I have commented my code in hard-to-understand areas.
- [x] I have made corresponding changes to the public documentation.
- [x] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [x] I have refactored the code for readability to the best of my ability.
- [x] I have checked that my changes do not introduce regressions in the public documentation.
- [x] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [x] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [x] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
I have a task for this in Jira
- [x] I have noted down and am currently tracking any technical debt introduced in this PR.
- [x] I have thoroughly thought about the architecture of this implementation.
- [x] I have asked peers to test this feature.

### CODE REVIEW

File: viewer/visual-tests/VisualTests.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright 2022 Cognite AS
+ */
+
+import { Page } from 'puppeteer';
+
+const environment = process?.env?.NODE_ENV ?? 'browser';
+
+if (environment === 'browser') {
```
Comment (larsmoa) [inline viewer/visual-tests/VisualTests.ts:9]: What is this if test about? Confused

**Review by larsmoa — COMMENTED** [review]
Looks very promising. Where will existing visual tests live? Thinking about how we can manage tests would be nice - but I'm not sure how we can make them "self registering".

File: viewer/visual-tests/VisualTests.ts (hunk 1 above)
Comment (christjt) [reply viewer/visual-tests/VisualTests.ts]: Was a dumb idea to have both the test runner and the server that serves the tests in the same file. And it turned out to just be confusing, so refactored to two different components.

**Comment by christjt** [comment]
> Looks very promising. Where will existing visual tests live? Thinking about how we can manage tests would be nice - but I'm not sure how we can make them "self registering".

Current visual tests will be distributed to relevant packages. Most will probably belong to /rendering package.

File: viewer/visual-tests/VisualTest.jest.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+/*!
+ * Copyright 2022 Cognite AS
+ */
+
+// import assert from 'assert';
```
Comment (pramodcog) [inline viewer/visual-tests/VisualTest.jest.ts:5]:
```suggestion
```

**Review by pramodcog — APPROVED** [review]

**Pull request merged**